server/queues/tasks/CommentUpdatedNotificationsTask.ts

```diff
@@ -53,6 +65,86 @@ export default class CommentUpdatedNotificationsTask extends BaseTask<CommentEve
           documentId: document.id,
           commentId: comment.id,
         });
+        userIdsMentioned.push(mention.modelId);
+      }
+    }
+  }
+
+  private async handleResolvedComment(event: CommentUpdateEvent) {
+    invariant(
+      !event.data?.newMentionIds,
+      "newMentionIds should not be present in resolved comment"
+    );
+
+    const [document, commentsAndReplies] = await Promise.all([
+      Document.scope("withCollection").findOne({
+        where: { id: event.documentId },
+      }),
+      Comment.findAll({
+        where: {
+          [Op.or]: [{ id: event.modelId }, { parentCommentId: event.modelId }],
+        },
+      }),
+    ]);
+
+    if (!document || !commentsAndReplies) {
+      return;
+    }
+
+    const send = async (userId: string) => {
+      await Notification.create({
+        event: NotificationEventType.ResolveComment,
+        userId,
+        actorId: event.actorId,
+        teamId: document.teamId,
+        documentId: document.id,
+        commentId: event.modelId,
+      });
+    };
+
+    const userIdsNotified: string[] = [];
+
+    // Don't notify resolver
+    userIdsNotified.push(event.actorId);
+
+    // Notify: (1) commenter (2) all repliers (3) all mentioned in the thread
+    for (const item of commentsAndReplies) {
+      if (!userIdsNotified.includes(item.createdById)) {
+        // -- author (comment or reply)
+        const user = await User.findByPk(item.createdById);
+
+        if (
+          user &&
+          user.id !== event.actorId &&
+          user.subscribedToEventType(NotificationEventType.ResolveComment) &&
+          (await canUserAccessDocument(user, document.id))
+        ) {
+          await send(user.id);
+          userIdsNotified.push(user.id);
+        }
+      }
+
+      // -- mentions
```

Comment:
Can we refactor this loop so that mentions is first. We can parse the mentioned user ID's, append the `createdById` to that array and then avoid having the rest of the logic twice here and the indirection to the `send` method.

Reply:
:flushed:

Reply:
I think it's an improvement? Less bug oppos

Reply:
Ah, I meant to say "I'm embarrassed for not doing that sooner" through that emoji 😄 .  So much cleaner.
I was fixated at retaining the copied logic from the mentions for some reason, and only factored out `send` :(